src/Language/PureScript/TypeChecker/TypeSearch.hs

```diff
@@ -76,6 +77,30 @@ checkSubsume unsolved env st userT envT = checkInEnvironment env st $ do
   -- Finally, check any constraints which were found during elaboration
   Entailment.replaceTypeClassDictionaries (isJust unsolved) expP
 
+accessorSearch
```

Comment:
Looks good, but is it possible to pull out some common logic here? Also, why don't we need the instance check?